### PR TITLE
Fix issue 18224 - BigInt modulo uint must return long.

### PR DESCRIPTION
**Pull request by markus-oberhumer**

### CODE REVIEW

File: std/bigint.d [hunk 1]
```diff
@@ -479,6 +479,7 @@ public:
 
         // BigInt % long => long
         // BigInt % ulong => BigInt
+        // BigInt % uint => long
```
Comment (wilzbach) [inline std/bigint.d]: What about the documentation table above?

File: std/bigint.d [hunk 2]
```diff
@@ -494,6 +495,14 @@ public:
                 return r;
             }
         }
+        else static if (is(Unqual!T == uint))
+        {
+            immutable uint u = absUnsign(y);
+            long rem = BigUint.modInt(data, u);
```
Comment (wilzbach) [inline std/bigint.d]: Why can't you join this with the else clause below?
AFAICT the only thing that is different is the type of `rem`:

```d
static if (static if (is(Unqual!T == uint))
    alias T = long;
else
    alias T = int;
```

**Comment by markus-oberhumer** [comment]
I've created a new version of this pull request to address the reviews above.

**Comment by markus-oberhumer** [comment]
Thanks!

**Pull request merged**